src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexPanel.tsx

```diff
@@ -590,3 +591,16 @@ function getSideBySideResetDisabledReason(index: IndexSharedInfo) {
 
     return null;
 }
+
+function getFaultyLocation(index: IndexSharedInfo, localNodeTag: string): databaseLocationSpecifier {
+    if (!IndexUtils.hasAnyFaultyNode(index)) {
+        return null;
+    }
+
+    const localFaultyInfo = index.nodesInfo.find((x) => x.details?.faulty && x.location.nodeTag === localNodeTag);
+    if (localFaultyInfo) {
+        return localFaultyInfo.location;
+    }
+
+    return index.nodesInfo.find((x) => x.details?.faulty)?.location;
```

Comment:
I wonder if this is a bit a magical. 
How about having an option to choose context (like we have for resetting index) or have this action available only if local node has faulty index?

Reply:
We already have "Open faulty index" for each location in details. Maybe the button in header should be visible only when local node is faulty
![image](https://github.com/user-attachments/assets/99cfb8a2-c8c9-49fc-9d74-319657599adf)

Reply:
yes, I forgot about that. Then let's only show it when local is faulty w/o fallback to other faulty nodes.